packages/flutter/lib/src/widgets/navigator_pop_handler.dart

```diff
@@ -33,14 +36,20 @@ import 'pop_scope.dart';
 ///    handle pops.
 ///  * [NavigationNotification], which indicates whether a [Navigator] in a
 ///    subtree can handle pops.
-class NavigatorPopHandler extends StatefulWidget {
+@optionalTypeArgs
+class NavigatorPopHandler<T> extends StatefulWidget {
   /// Creates an instance of [NavigatorPopHandler].
   const NavigatorPopHandler({
     super.key,
+    @Deprecated(
+      'Use onPopWithResult instead. '
+      'This feature was deprecated after v3.26.0-0.1.pre.',
+    )
     this.onPop,
+    this.onPopWithResult,
```

Comment:
I vaguely remember i didn't migrate this widget is that i can't find a use case where someone would like to handle result, but I guess you could use have use case if you call maybe pop on the parent navigator?

Reply:
@math1man is that how you are using this?